cmd/influx_inspect/export/export.go

```diff
@@ -0,0 +1,417 @@
+		db, rp := influxql.QuoteIdent(keys[0]), influxql.QuoteIdent(keys[1])
+		fmt.Fprintf(w, "CREATE DATABASE %s WITH NAME %s\n", db, rp)
+	}
+
+	fmt.Fprintln(w, "# DML")
+	for key := range cmd.manifest {
+		keys := strings.Split(key, string(byte(os.PathSeparator)))
+		fmt.Fprintf(w, "# CONTEXT-DATABASE:%s\n", keys[0])
+		fmt.Fprintf(w, "# CONTEXT-RETENTION-POLICY:%s\n", keys[1])
+		if files, ok := cmd.tsmFiles[key]; ok {
+			fmt.Printf("writing out tsm file data for %s...", key)
+			if err := cmd.writeTsmFiles(w, files); err != nil {
+				return err
+			}
+			fmt.Println("complete.")
+		}
+		if _, ok := cmd.walFiles[key]; ok {
+			fmt.Printf("writing out wal file data for %s...", key)
+			if err := cmd.writeWALFiles(w, cmd.walFiles[key], key); err != nil {
+				return err
+			}
+			fmt.Println("complete.")
+		}
+	}
+	return nil
+}
+
+func (cmd *Command) writeTsmFiles(w io.WriteCloser, files []string) error {
+	fmt.Fprintln(w, "# writing tsm data")
+
+	// we need to make sure we write the same order that the files were written
+	sort.Strings(files)
+
+	// use a function here to close the files in the defers and not let them accumulate in the loop
+	write := func(f string) error {
+		file, err := os.OpenFile(f, os.O_RDONLY, 0600)
+		if err != nil {
+			return fmt.Errorf("%v", err)
+		}
+		defer file.Close()
+		reader, err := tsm1.NewTSMReader(file)
+		if err != nil {
+			log.Printf("unable to read %s, skipping\n", f)
+			return nil
+		}
+		defer reader.Close()
+
+		if sgStart, sgEnd := reader.TimeRange(); sgStart > cmd.endTime || sgEnd < cmd.startTime {
+			return nil
+		}
+
+		for i := 0; i < reader.KeyCount(); i++ {
+			var pairs string
+			key, typ := reader.KeyAt(i)
+			values, _ := reader.ReadAll(string(key))
+			measurement, field := tsm1.SeriesAndFieldFromCompositeKey(key)
+
+			for _, value := range values {
+				if (value.UnixNano() < cmd.startTime) || (value.UnixNano() > cmd.endTime) {
+					continue
+				}
+
+				switch typ {
+				case tsm1.BlockFloat64:
+					pairs = field + "=" + fmt.Sprintf("%v", value.Value())
+				case tsm1.BlockInteger:
+					pairs = field + "=" + fmt.Sprintf("%vi", value.Value())
+				case tsm1.BlockBoolean:
+					pairs = field + "=" + fmt.Sprintf("%v", value.Value())
+				case tsm1.BlockString:
+					pairs = field + "=" + fmt.Sprintf("%q", models.EscapeStringField(fmt.Sprintf("%s", value.Value())))
+				default:
+					pairs = field + "=" + fmt.Sprintf("%v", value.Value())
+				}
+
+				fmt.Fprintln(w, string(measurement), pairs, value.UnixNano())
+			}
+		}
+		return nil
+	}
+
+	for _, f := range files {
+		if err := write(f); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (cmd *Command) writeWALFiles(w io.WriteCloser, files []string, key string) error {
+	fmt.Fprintln(w, "# writing wal data")
+
+	// we need to make sure we write the same order that the wal received the data
+	sort.Strings(files)
+
+	var once sync.Once
+	warn := func() {
+		msg := fmt.Sprintf(`WARNING: detected deletes in wal file.
```

Comment:
nit, you could just do:

```go
msg := `foo blah %q blah \n`
fmt.Fprintf(cmd.Stderr, msg, key)
```

It's seems a little tidier than calling `fmt.Sprintf` and then `fmt.Fprintln`